src/python/pants/engine/legacy/address_mapper.py

```diff
@@ -67,12 +66,10 @@ def _internal_scan_specs(self, specs, fail_fast=True, missing_is_fatal=True):
     # TODO: This should really use `product_request`, but on the other hand, we need to
     # deprecate the entire `AddressMapper` interface anyway. See #4769.
     request = self._scheduler.execution_request([BuildFileAddresses], [Specs(tuple(specs))])
-    result = self._scheduler.execute(request)
-    if result.error:
-      raise self.BuildFileScanError(str(result.error))
-    (_, state), = result.root_products
+    returns, throws = self._scheduler.execute(request)
 
-    if isinstance(state, Throw):
+    if throws:
+      _, state = throws[0]
```

Comment:
line 83 below is raising, you want to initialize `state`: 
_, state = returns[0]

Reply:
Thanks, fixed!